### PR TITLE
Shooting railgun pattern

### PR DESCRIPTION
**Pull request by Alexwen709**
All of rongrui's changes including the railgun pattern (pattern used for shooting note)

### CODE REVIEW

**Comment by Alexwen709** [comment]
currently fixing merge conflicts due to a lot of changes

**Comment by Alexwen709** [comment]
update: 
merge conflicts should be fixed

File: robot_states.py [hunk 1]
```diff
@@ -1,14 +1,19 @@
 import time
 from light import Light
-frames = 25
-frame_interval = 1
+frames = 70
+frame_interval = 5
 frame_time_per_interval = frame_interval / frames
 
 recorded_mode = 0
+starting = time.time()
```
Comment (aschokking) [inline robot_states.py]: ⭐ unused?

File: main.py [hunk 2]
```diff
@@ -44,14 +44,26 @@ def match_state_with_pattern(light: Light):
...
     # Also, "action" or "rainbow" states should have higher priority than static
     global robot_state
     global recorded_mode
     patterns.loopcount += 1
-    robot_state = get_states() # Placed here because we always want to update our states
+
+    new_states = get_states()
+    for light in lights:
+        if new_states[light.channel] != robot_state[light.channel]:
+            print("CHanged")
```
Comment (aschokking) [inline main.py:56]: ⭐ nit typo, cap H

**Review by aschokking — APPROVED** [review]

**Pull request merged**